packages/compiler-cli/ngcc/src/packages/entry_point_manifest.ts

```diff
@@ -0,0 +1,139 @@
+   * `basePath` or `null` if the manifest was out of date.
+   */
+  readEntryPointsUsingManifest(basePath: AbsoluteFsPath): EntryPoint[]|null {
+    try {
+      if (this.fs.basename(basePath) !== 'node_modules') {
+        return null;
+      }
+
+      const computedLockFileHash = this.computeLockFileHash(basePath);
+      if (computedLockFileHash === null) {
+        return null;
+      }
+
+      const manifestPath = this.getEntryPointManifestPath(basePath);
+      if (!this.fs.exists(manifestPath)) {
+        return null;
+      }
+
+      const {ngccVersion, configFileHash, lockFileHash, entryPointPaths} =
+          JSON.parse(this.fs.readFile(manifestPath)) as EntryPointManifestFile;
+      if (ngccVersion !== NGCC_VERSION || configFileHash !== this.config.hash ||
+          lockFileHash !== computedLockFileHash) {
+        return null;
+      }
+
+      this.logger.debug(
+          `Entry-point manifest found for ${basePath} so loading entry-point information directly.`);
+      const startTime = Date.now();
+
+      const entryPoints: EntryPoint[] = [];
+      for (const [packagePath, entryPointPath] of entryPointPaths) {
+        const result =
+            getEntryPointInfo(this.fs, this.config, this.logger, packagePath, entryPointPath);
+        if (result === NO_ENTRY_POINT || result === INVALID_ENTRY_POINT) {
+          throw new Error(
+              `The entry-point manifest at ${manifestPath} contained an invalid pair of package paths: [${packagePath}, ${entryPointPath}]`);
+        } else {
+          entryPoints.push(result);
+        }
+      }
+      const duration = Math.round((Date.now() - startTime) / 100) / 10;
+      this.logger.debug(`Reading entry-points directly took ${duration}s.`);
+      return entryPoints;
+    } catch (e) {
+      this.logger.warn(`Unable to read the entry-point manifest for ${basePath}`, e.toString());
+      return null;
+    }
+  }
+
+  /**
+   * Write a manifest file at the given `basePath`.
+   *
+   * The manifest includes the current ngcc version and hashes of the package lock-file and current
+   * project config. These will be used to check whether the manifest file is out of date. See
+   * `readEntryPointsUsingManifest()`.
+   *
+   * @param basePath The path where the manifest file is to be written.
+   * @param entryPoints A collection of entry-points to record in the manifest.
+   */
+  writeEntryPointManifest(basePath: AbsoluteFsPath, entryPoints: EntryPoint[]): void {
+    const lockFileHash = this.computeLockFileHash(basePath);
+    if (lockFileHash === null) {
+      return;
+    }
+    const manifest: EntryPointManifestFile = {
+      ngccVersion: NGCC_VERSION,
+      configFileHash: this.config.hash,
+      lockFileHash: lockFileHash,
+      entryPointPaths: entryPoints.map(entryPoint => [entryPoint.package, entryPoint.path]),
+    };
+    this.fs.writeFile(this.getEntryPointManifestPath(basePath), JSON.stringify(manifest));
```

Comment:
Consider writing this as readable JSON:
```suggestion
    this.fs.writeFile(this.getEntryPointManifestPath(basePath), JSON.stringify(manifest, null, 2));
```

Reply:
I avoided this because:

a) It is not supposed to be read by humans. And if they want to read it, it is trivial to format it in an editor
b) These files can get quite large. Adding extra whitespace slows down the stringification/parsing for now programmatic benefit. Given that this change is supposed to be for performance reasons that seemed a strange thing to do.